### PR TITLE
GH-35445: [R] Behavior something like group_by(foo) |> across(everything()) is different from dplyr

### LINKED ISSUE #35445: [R] Behavior something like `group_by(foo) |> across(everything())` is different from dplyr
### Describe the bug, including details regarding any error messages, version, and platform.

In dplyr, I believe that using `across(everything())` on a grouped data frame will not select the column used for grouping.

``` r
mtcars |>
  dplyr::group_by(cyl) |>
  dplyr::summarise(dplyr::across(everything(), sum))
#> # A tibble: 3 × 11
#>     cyl   mpg  disp    hp  drat    wt  qsec    vs    am  gear  carb
#>   <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
#> 1     4  293. 1156.   909  44.8  25.1  211.    10     8    45    17
#> 2     6  138. 1283.   856  25.1  21.8  126.     4     3    27    24
#> 3     8  211. 4943.  2929  45.2  56.0  235.     0     2    46    49
```

<sup>Created on 2023-05-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

However, arrow does not seem to exclude the columns used for grouping. The following example results in an error.
(I installed arrow 12.0.0.20230503 from R-universe)

``` r
mtcars |>
  arrow::as_arrow_table() |>
  dplyr::group_by(cyl) |>
  dplyr::summarise(dplyr::across(everything(), sum)) |>
  dplyr::collect()
#> Error in `compute.arrow_dplyr_query()`:
#> ! Invalid: Multiple matches for FieldRef.Name(cyl) in mpg: double
#> cyl: double
#> disp: double
#> hp: double
#> drat: double
#> wt: double
#> qsec: double
#> vs: double
#> am: double
#> gear: double
#> carb: double
#> cyl: double
#> Backtrace:
#>     ▆
#>  1. ├─dplyr::collect(...)
#>  2. └─arrow:::collect.arrow_dplyr_query(...)
#>  3.   └─arrow:::compute.arrow_dplyr_query(x)
#>  4.     └─base::tryCatch(...)
#>  5.       └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>  6.         └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>  7.           └─value[[3L]](cond)
#>  8.             └─arrow:::augment_io_error_msg(e, call, schema = schema())
#>  9.               └─rlang::abort(msg, call = call)
```

<sup>Created on 2023-05-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### Component(s)

R

### PR DESCRIPTION
**Pull request by eitsupi**
### Rationale for this change

The argument `.cols` of the `dplyr::across` function has the following description.

> You can't select grouping columns because they are already automatically handled by the verb (i.e. summarise() or mutate()).

However, this behavior is currently not reproduced in the `arrow` package and an error occurs when selecting the column used for grouping with `everything()`.

``` r
mtcars |>
  arrow::as_arrow_table() |>
  dplyr::group_by(cyl) |>
  dplyr::summarise(dplyr::across(everything(), sum)) |>
  dplyr::collect()
#> Error in `compute.arrow_dplyr_query()`:
#> ! Invalid: Multiple matches for FieldRef.Name(cyl) in mpg: double
#> cyl: double
#> disp: double
#> hp: double
#> drat: double
#> wt: double
#> qsec: double
#> vs: double
#> am: double
#> gear: double
#> carb: double
#> cyl: double
#> Backtrace:
#>     ▆
#>  1. ├─dplyr::collect(...)
#>  2. └─arrow:::collect.arrow_dplyr_query(...)
#>  3.   └─arrow:::compute.arrow_dplyr_query(x)
#>  4.     └─base::tryCatch(...)
#>  5.       └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>  6.         └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>  7.           └─value[[3L]](cond)
#>  8.             └─arrow:::augment_io_error_msg(e, call, schema = schema())
#>  9.               └─rlang::abort(msg, call = call)
```

<sup>Created on 2023-05-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

This PR fixes this behavior to match with dplyr's original behavior.

### What changes are included in this PR?

- Auto exclude grouping columns in `across` in `mutate`, `transmute`, and `summarise`.
- The `.data` argument of internal function `expand_across` should be `arrow_dplyr_query`.
  Some tests have been slightly modified to accommodate this change.
- `mutate`, `transmute`, `arrange`, `filter` always return `arrow_dplyr_query`.
  Currently, `arrow_dplyr_query` is not returned in the following cases, which was not consistent. 
  ```r
  mtcars |> arrow::arrow_table() |> dplyr::mutate()
  ```
- Correct the order of columns in results of `group_by(foo) |> mutate(.keep = "none")`
  Currently, the results of the following query show that the columns used for grouping have moved to the tail and differ from the behavior of dplyr.
  ```r
  mtcars |> arrow::arrow_table() |> dplyr::group_by(cyl) |> dplyr::mutate(am, .keep = "none") |> dplyr::collect()
  ```
- Correct the order of columns in results of `group_by(foo) |> transmute()`
  Currently, the results of the following query show that the columns used for grouping have moved to the tail and differ from the behavior of dplyr.
  ```r
  mtcars |> arrow::arrow_table() |> dplyr::group_by(cyl) |> dplyr::transmute(mpg) |> dplyr::collect()
  ```
  After `transmute`, the group columns should move to the left. (This is a different behavior from `mutate(.keep = "none")`, which keeps the original position.)

### Are these changes tested?

Yes.

### Are there any user-facing changes?

Yes.

* Closes: #35445

### CODE REVIEW

**Review by thisisnic — COMMENTED** [review]
Thanks for catching this! Would you mind updating the tests to use the `example_data` dataset, for consistency with other tests?

**Comment by eitsupi** [comment]
I noticed that the column order is also wrong after `select`, but this is beyond the scope of this pull request, so I will create another issue.

```r
> mtcars |> group_by(cyl) |> select(mpg) |> collect()
Adding missing grouping variables: `cyl`
# A tibble: 32 × 2
# Groups:   cyl [3]
     cyl   mpg
   <dbl> <dbl>
 1     6  21  
 2     6  21  
 3     4  22.8
 4     6  21.4
 5     8  18.7
 6     6  18.1
 7     8  14.3
 8     4  24.4
 9     4  22.8
10     6  19.2
# … with 22 more rows
# ℹ Use `print(n = ...)` to see more rows

> mtcars |> arrow_table() |> group_by(cyl) |> select(mpg) |> collect()
# A tibble: 32 × 2
# Groups:   cyl [3]
     mpg   cyl
   <dbl> <dbl>
 1  21       6
 2  21       6
 3  22.8     4
 4  21.4     6
 5  18.7     8
 6  18.1     6
 7  14.3     8
 8  24.4     4
 9  22.8     4
10  19.2     6
# … with 22 more rows
# ℹ Use `print(n = ...)` to see more rows
```

File: r/tests/testthat/test-dplyr-mutate.R [hunk 1]
```diff
@@ -652,3 +672,41 @@ test_that("Can use across() within transmute()", {
...
 })
+
+test_that("across() does not select grouping variables within mutate()", {
+  compare_dplyr_binding(
+    .input %>%
+      group_by(chr) %>%
+      mutate(across(everything(), round)) %>%
+      collect(),
+    example_data %>%
+      select(int, dbl, chr)
```
Comment (thisisnic) [inline r/tests/testthat/test-dplyr-mutate.R:683]: nit: it's more skimmable when the `tbl` parameter to `compare_dplyr_binding()` is just a table with no modification. Please could you move the `select()` to the `expr` parameter code instead?

**Review by thisisnic — APPROVED** [review]
One tiny change to the tests, otherwise this is good to go.  Thanks!

**Pull request merged**